Disable `filenames/match-exported` rule for generated `config/environment.d.ts` in `ember` config

raycohen opened the pull request:
This rule triggers in CI after creating a new addon in Dashboard. [Example build part](https://kochiku-production-build-logs.s3.us-west-2.amazonaws.com/squareup/dashboard/build_6321230/part_322014409/attempt_277137685/stdout.log.gz?X-Amz-Expires=604800&X-Amz-Date=20221205T160212Z&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIBPRFGE3IOTXUBCA%2F20221205%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-SignedHeaders=host&X-Amz-Signature=069d5b3769ded56a17e1120a7680ce05e0f9f39660ad750db6e428ca7b0e3b75).

```
/data/app/kochiku-worker/shared/build-partition/squareup/dashboard/frontend/@dashboard/ember-cli-csso/tests/dummy/app/config/environment.d.ts
  5:1  error  Filename 'environment' must match the exported and transformed name 'config'  filenames/match-exported
```


The generated code is defined [here](https://github.com/typed-ember/ember-cli-typescript/blob/bb6754670e7fe550516998e2965fff4097fcb2c3/blueprint-files/ember-cli-typescript/__config_root__/config/environment.d.ts) in ember-cli-typescript.

Related to https://github.com/square/eslint-plugin-square/pull/142

bmish commented:
I wonder if it would make sense to rename the variable in ember-cli-typescript to fix the lint violation? It's a tough case because the variable name `config` probably makes a bit more sense than `environment` in this case. Note that I have fixed the blueprint in ember-cli-typescript to avoid lint violations for us several times so it is an option available to us if it makes sense ever.

bmish commented on lib/config/ember.js:
```diff
@@ -52,6 +52,7 @@ module.exports = {
     {
       files: [
         '**/{app,addon}/{components,controllers,routes,services}/**/*.{js,ts}',
+        '**/dummy/app/config/environment.d.ts',
```
How about we create a separate override to turn off the rule for each set of files? That way, we can have a helpful comment for each override, instead of this current situation where the comment only applies to one of the items in the files array.

raycohen commented:
> I wonder if it would make sense to rename the variable in ember-cli-typescript to fix the lint violation? It's a tough case because the variable name config probably makes a bit more sense than environment in this case. Note that I have fixed the blueprint in ember-cli-typescript to avoid lint violations for us several times so it is an option available to us if it makes sense ever.

I updated to give it its own override. 

Regarding updating the blueprint - for what it's worth it looks like the js [version](https://github.com/ember-cli/ember-cli/blob/1e415adb425fe219f4d8708be64c792be81f7385/blueprints/app/files/config/environment.js#L4) of this blueprint calls the variable `ENV`. I feel like `environment` isn't a great name for the object though, and `config` or `environmentConfig` are better fits, especially because it has an `environment` key inside the config. Don't feel too strongly though.

bmish reviewed (approved):
Thanks!
